src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -155,32 +181,95 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
     @Override
     protected Position handlePosition(Position position) {
 
-        String url;
-        if (json) {
-            url = this.url;
-        } else {
-            try {
-                url = formatRequest(position);
-            } catch (UnsupportedEncodingException | JsonProcessingException e) {
-                throw new RuntimeException("Forwarding formatting error", e);
+        class AsyncRequestAndCallback implements InvocationCallback<Response> {
+
+            private int delay = retryDelayMin;
+            private Map<String, Object> payload;
+            private Invocation.Builder requestBuilder;
+
+            AsyncRequestAndCallback(Position position) {
+
+                String formattedUrl;
+                try {
+                    formattedUrl = json ? url : formatRequest(position);
+                } catch (UnsupportedEncodingException | JsonProcessingException e) {
+                    throw new RuntimeException("Forwarding formatting error", e);
+                }
+
+                requestBuilder = client.target(formattedUrl).request();
+                if (header != null && !header.isEmpty()) {
+                    for (String line: header.split("\\r?\\n")) {
+                        String[] values = line.split(":", 2);
+                        requestBuilder.header(values[0].trim(), values[1].trim());
+                    }
+                }
+
+                if (json) {
+                    payload = prepareJsonPayload(position);
+                }
+
+                deliveryPendingCurrent.incrementAndGet();
+
+                send();
             }
-        }
 
-        Invocation.Builder requestBuilder = client.target(url).request();
+            private void send() {
+                if (json) {
+                    requestBuilder.async().post(Entity.json(payload), this);
+                } else {
+                    requestBuilder.async().get(this);
+                }
+            }
 
-        if (header != null && !header.isEmpty()) {
-            for (String line: header.split("\\r?\\n")) {
-                String[] values = line.split(":", 2);
-                requestBuilder.header(values[0].trim(), values[1].trim());
+            private void retry() {
+                try {
+                    String message = "Position forwarding failed.";
```

Comment:
Doesn't make sense to have a variable for logging message. Duplicate it if needed, but I would just make all logging less verbose. Make text shorter.

Reply:
Maybe we just need to notify about the error and print how many positions are pending.
Something like:
Position forwarding failed: 10 pending.
Would this be ok?

Reply:
Sounds good to me. We don't need a lot of logging. Also don't add "." at the end.

Reply:
Good. Done.